### PR TITLE
[all] Minor code cleaning

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 1]
```diff
@@ -117,15 +117,15 @@ void GenericConstraintCorrection::bwdInit()
...
     {
         msg_error() << "No LinearSolver found.";
         return;
     }
 
     msg_info() << "Found " << m_linearSolvers.size() << " m_linearSolvers";
-    for (unsigned i = 0; i < m_linearSolvers.size(); i++)
-        msg_info() << m_linearSolvers[i]->getName();
+    for (auto & m_linearSolver : m_linearSolvers)
+        msg_info() << m_linearSolver->getName();
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
        msg_info() << linearSolver->getName();
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 2]
```diff
@@ -150,8 +150,8 @@ void GenericConstraintCorrection::removeConstraintSolver(ConstraintSolver *s)
 
 void GenericConstraintCorrection::rebuildSystem(double massFactor, double forceFactor)
 {
-    for (unsigned i = 0; i < m_linearSolvers.size(); i++)
-        m_linearSolvers[i]->rebuildSystem(massFactor, forceFactor);
+    for (auto & m_linearSolver : m_linearSolvers)
+        m_linearSolver->rebuildSystem(massFactor, forceFactor);
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
        linearSolver->rebuildSystem(massFactor, forceFactor);
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 3]
```diff
@@ -180,17 +180,17 @@ void GenericConstraintCorrection::addComplianceInConstraintSpace(const Constrain
 
     factor *= complianceFactor;
     // use the Linear solver to compute J*inv(M)*Jt, where M is the mechanical linear system matrix
-    for (unsigned i = 0; i < m_linearSolvers.size(); i++)
+    for (auto & m_linearSolver : m_linearSolvers)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 4]
```diff
@@ -180,17 +180,17 @@ void GenericConstraintCorrection::addComplianceInConstraintSpace(const Constrain
...
     {
-        m_linearSolvers[i]->buildComplianceMatrix(cparams, W, factor);
+        m_linearSolver->buildComplianceMatrix(cparams, W, factor);
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
        linearSolver->buildComplianceMatrix(cparams, W, factor);
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 5]
```diff
@@ -180,17 +180,17 @@ void GenericConstraintCorrection::addComplianceInConstraintSpace(const Constrain
...
     }
 }
 
 void GenericConstraintCorrection::computeMotionCorrectionFromLambda(const ConstraintParams* cparams, MultiVecDerivId dx, const defaulttype::BaseVector * lambda)
 {
-    for (Size i = 0; i < m_linearSolvers.size(); ++i)
+    for (auto & m_linearSolver : m_linearSolvers)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 6]
```diff
@@ -180,17 +180,17 @@ void GenericConstraintCorrection::addComplianceInConstraintSpace(const Constrain
...
     {
-        m_linearSolvers[i]->applyConstraintForce(cparams, dx, lambda);
+        m_linearSolver->applyConstraintForce(cparams, dx, lambda);
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
        linearSolver->applyConstraintForce(cparams, dx, lambda);
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 7]
```diff
@@ -202,10 +202,10 @@ void GenericConstraintCorrection::applyMotionCorrection(const ConstraintParams*
                                                         double positionFactor,
                                                         double velocityFactor)
 {
-    for (Size i = 0; i < m_linearSolvers.size(); ++i)
+    for (auto & m_linearSolver : m_linearSolvers)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 8]
```diff
@@ -202,10 +202,10 @@ void GenericConstraintCorrection::applyMotionCorrection(const ConstraintParams*
...
     {
-        MechanicalIntegrateConstraintsVisitor v(cparams, positionFactor, velocityFactor, correction, dxId, xId, vId, m_linearSolvers[i]->getSystemMultiMatrixAccessor());
-        m_linearSolvers[i]->getContext()->executeVisitor(&v);
+        MechanicalIntegrateConstraintsVisitor v(cparams, positionFactor, velocityFactor, correction, dxId, xId, vId, m_linearSolver->getSystemMultiMatrixAccessor());
+        m_linearSolver->getContext()->executeVisitor(&v);
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
        MechanicalIntegrateConstraintsVisitor v(cparams, positionFactor, velocityFactor, correction, dxId, xId, vId, linearSolver->getSystemMultiMatrixAccessor());
        linearSolver->getContext()->executeVisitor(&v);
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 9]
```diff
@@ -262,9 +262,9 @@ void GenericConstraintCorrection::applyContactForce(const BaseVector *f)
 
 void GenericConstraintCorrection::computeResidual(const ExecParams* params, defaulttype::BaseVector *lambda)
 {
-    for (unsigned i = 0; i < m_linearSolvers.size(); i++)
+    for (auto & m_linearSolver : m_linearSolvers)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
    for (auto & linearSolver : m_linearSolvers)
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp [hunk 10]
```diff
@@ -262,9 +262,9 @@ void GenericConstraintCorrection::applyContactForce(const BaseVector *f)
...
     {
-        m_linearSolvers[i]->computeResidual(params, lambda);
+        m_linearSolver->computeResidual(params, lambda);
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintCorrection.cpp]:
```suggestion
        linearSolver->computeResidual(params, lambda);
```

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 11]
```diff
@@ -628,12 +622,12 @@ void GenericConstraintProblem::clear(int nbC)
 
 void GenericConstraintProblem::freeConstraintResolutions()
 {
-    for(unsigned int i=0; i<constraintsResolutions.size(); i++)
+    for(auto & constraintsResolution : constraintsResolutions)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: Reference to a pointer ?

**Review by fredroy — CHANGES_REQUESTED** [review]

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp (hunk 11 above)
Comment (alxbilger) [reply modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: good catch

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 12]
```diff
@@ -137,15 +135,15 @@ void GenericConstraintSolver::init()
     // Prevents ConstraintCorrection accumulation due to multiple AnimationLoop initialization on dynamic components Add/Remove operations.
     if (!constraintCorrections.empty())
     {
-        for (unsigned int i = 0; i < constraintCorrections.size(); i++)
-            constraintCorrections[i]->removeConstraintSolver(this);
+        for (auto & constraintCorrection : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: Actually this file has a lot of cases where it takes now a ref to a pointer

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 13]
```diff
@@ -137,15 +135,15 @@ void GenericConstraintSolver::init()
...
+            constraintCorrection->removeConstraintSolver(this);
         constraintCorrections.clear();
     }
 
     getContext()->get<core::behavior::BaseConstraintCorrection>(&constraintCorrections, core::objectmodel::BaseContext::SearchDown);
     constraintCorrectionIsActive.resize(constraintCorrections.size());
-    for (unsigned int i = 0; i < constraintCorrections.size(); i++)
-        constraintCorrections[i]->addConstraintSolver(this);
+    for (auto & constraintCorrection : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: here...

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 14]
```diff
@@ -167,8 +165,8 @@ void GenericConstraintSolver::cleanup()
 {
     if (!constraintCorrections.empty())
     {
-        for (unsigned int i = 0; i < constraintCorrections.size(); i++)
-            constraintCorrections[i]->removeConstraintSolver(this);
+        for (auto & constraintCorrection : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: here...

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 15]
```diff
@@ -260,9 +258,8 @@ bool GenericConstraintSolver::buildSystem(const core::ConstraintParams *cParams,
 
     if (unbuilt.getValue())
     {
-        for (unsigned int i=0;i<constraintCorrections.size();i++)
+        for (auto cc : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: and from here, I guess it should be auto* cc ? ( but I guess in this case it does not matter actually)

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 16]
```diff
@@ -354,9 +351,8 @@ bool GenericConstraintSolver::buildSystem(const core::ConstraintParams *cParams,
             }
 
         } else {
-            for (unsigned int i=0; i<constraintCorrections.size(); i++)
+            for (auto cc : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: bis

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 17]
```diff
@@ -381,9 +377,8 @@ bool GenericConstraintSolver::buildSystem(const core::ConstraintParams *cParams,
 
 void GenericConstraintSolver::rebuildSystem(double massFactor, double forceFactor)
 {
-    for (unsigned int i=0; i<constraintCorrections.size(); i++)
+    for (auto cc : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: ter

File: modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp [hunk 18]
```diff
@@ -484,9 +479,8 @@ bool GenericConstraintSolver::solveSystem(const core::ConstraintParams * /*cPara
 
 void GenericConstraintSolver::computeResidual(const core::ExecParams* eparam)
 {
-    for (unsigned int i=0; i<constraintCorrections.size(); i++)
+    for (auto cc : constraintCorrections)
```
Comment (fredroy) [inline modules/SofaConstraint/src/SofaConstraint/GenericConstraintSolver.cpp]: quater

**Review by fredroy — CHANGES_REQUESTED** [review]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
seems like it introduces two new failed unit test:
 - UnitTests.Compliant_test/Batch/Python_scene_test.sofa_python_scene_tests/2
 - UnitTests.SofaCarving_test/SofaCarving_test.testdoCarving
 
 I suppose investigating the SofaCarving test will fix the Compliant one

**Review by fredroy — APPROVED** [review]
Great, tests are passing 👍

**Pull request merged**